src/upnp.cpp

```diff
@@ -227,6 +232,11 @@ port_mapping_t upnp::add_mapping(portmap_protocol const p, int const external_po
 
 	if (mapping_it == m_mappings.end())
 	{
+		if (m_mappings.end_index() >= max_global_mappings)
+		{
+			log("too many mappings registered");
```

Comment:
do you think it makes sense to add an assert here too? This isn't really expected to happen, right? (I suppose the client can add its own mappings and exceed this limit, but it doesn't seem all that likely)

Reply:
yes, I will the assert